File: Composition/CS/Composition/Algorithm/Composition/AveragingStrategy.cs
using System.Collections.Generic;
using System.Linq;

namespace Algorithm.Composition
{
    public class AveragingStrategy : IAggregationStrategy
    {
        public Measurement Aggregate(IEnumerable<Measurement> measurements)
        {
            return new Measurement
                       {
                           X = (int)measurements.Average(m => m.X),
                           Y = (int)measurements.Average(m => m.Y)
                       };
        }
    }
}

File: Composition/Java/Composition/src/main/java/algorithm/composition/AveragingStrategy.java
package algorithm.composition;

import java.util.Collection;

import algorithm.Measurement;
import algorithm.MeasurementUtil;

public class AveragingStrategy implements IAggregationStrategy {

	@Override
	public Measurement aggregate(Collection<Measurement> measurements) {
		return new Measurement(MeasurementUtil.averageX(measurements), MeasurementUtil.averageY(measurements));
	}

}
